### ISSUE TITLE
Error using vg construct: Index is older than VCF (though it's not)

### ISSUE THREAD

**Issue by maxineliu** [vgteam/vg#3686]
**1. What were you trying to do?**
I'm tring to construct reference graph `bufo.vg` using reference `genomic.fna`, genome index `genomic.fna.fai` and vcf `bufo.12samples.vcf.gz`, vcf index `bufo.12samples.vcf.gz.csi`. I have to generate `.csi` index instead of `.tbi` because the bufo chromosomes are too big to be stored in `tbi` index. I'm pretty sure that bufo.12samples.vcf.gz.csi is younger than bufo.12samples.vcf.gz.
```
-rw-r----- 1 maxine91 maxine91 1.1G Jun 14 23:15 bufo.12samples.vcf.gz
-rw-r----- 1 maxine91 maxine91 631K Jun 15 13:22 bufo.12samples.vcf.gz.csi
```

**2. What did you want to happen?**
build `bufo.vg`

**3. What actually happened?**
```
[tabix++] the index file is older than the vcf file. Please use '-f' to overwrite or reindex.
```

**5. What data and command can the vg dev team use to make the problem happen?**
```
./vg construct -r genomic.fna -v bufo.12samples.vcf.gz -t 24 -m 32  > bufo.vg
```

**6. What does running `vg version` say?**

```
vg: variation graph tool, version v1.41.0 "Salmour"
```

**Comment by glennhickey** [comment]
I'm pretty sure that `[tabix++] the index file is older than the vcf file. Please use '-f' to overwrite or reindex.` is just a an error that vg gives when it can't find the `.tbi`.  So your issue seems to reveal two problems:
* the error message is confusing when vg can't find the vcf index
* vg does not support `.csi` VCF indexes.

**Comment by maxineliu** [comment]
According to [an answer](https://www.biostars.org/p/9527307/#9527323:~:text=they%20seem%20to%20check%20for%20a%20TBI%20or%20CSI%20file%20(https%3A//github.com/vgteam/vg/blob/2c36d2cd08bef668eca2a8a64ba51abc037c905b/src/index_registry.cpp%23L561)%20before%20the%20error%20message%20though%2C%20so%20I%20don%27t%20know%20what%27s%20happening.) on biostars forum, vg seems trying to check if csi exist before the error msg.

**Comment by glennhickey** [comment]
Yeah, so  [biostars points out](https://www.biostars.org/p/9527307/#9527319) there's a previous issue about the confusing error message: #1414.   

The part about vg not supporting csi indexes seems new to this issue...

**Comment by maxineliu** [comment]
Yes. no idea if this issue appears results from vg not supporting csi. If so, that is really bad news for me.

**Comment by glennhickey** [comment]
**vg does not support csi**

**Comment by maxineliu** [comment]
Is there a plan to let vg supports csi? Or Any suggestions for my case?

**Comment by glennhickey** [comment]
Your best bet in the very short term would be to split your chromosome manually.   

I think any eventual support in vg for .csi indexes would have to come by way of [vcflib](https://github.com/vcflib/vcflib).  You can try asking at that repo, as it seems to be actively being maintained.  

Otherwise, I don't think this is too high on vgteam's radar right now though obviously it'd be nice to support eventually.

**Comment by faithokamoto** [comment]
The error message is now e.g.
```
[E::idx_find_and_load] Could not retrieve index file for 'x.vcf.gz'
[tabix++] failed to load the index file.
```
and the CSI support thing is tracked in #3864, so I'm closing this issue.
